ckanext/canada/helpers.py

```diff
@@ -774,6 +776,52 @@ def date_field(field, pkg):
     return pkg.get(field, None)
 
 
+def goc_auth():
+    if not registry_network_access():
+        return t.abort(403, _(u'This application is only available to authorized '
+                            u'Government of Canada departments and agencies. '
+                            u'Please contact the support team at '
+                            u'open-ouvert@tbs-sct.gc.ca to request access.'))
+
+
+def registry_network_access():
+    """
+    Only allow requests from GOC network to access
+    user account registration view
+   """
+    remote_addr = request.headers.get('X-Forwarded-For') or \
+                  request.environ.get('REMOTE_ADDR')
+    try:
+        client_ip = ipaddress.ip_address(unicode(remote_addr))
```

Comment:
@RabiaSajjad Can you also change the usage of `unicode` to `text_type` so this is Python3 ready. I think the `text_type` from `six` should already be imported in the helpers here.